src/apps/slackapp/tests/test_get_response.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+
+import pytest
+from slackapp.bolt_app import get_response
+
+import sherpa_ai.config as cfg
+from sherpa_ai.verbose_loggers import DummyVerboseLogger
+
+
+def test_get_date():
+    question = "What is the date today, using the following format: YYYY-MM-DD?"
+    date = datetime.now().strftime("%Y-%m-%d")
+
+    if cfg.SERPER_API_KEY is None:
+        pytest.skip(
+            "SERPER_API_KEY not found in environment variables, skipping this test"
+        )
+
+    verbose_logger = DummyVerboseLogger()
+
+    response = get_response(
+        question=question,
+        previous_messages=[],
+        user_id="",
+        team_id="",
+        verbose_logger=verbose_logger,
+        bot_dict={"user_id": "Sherpa"},
+    )
+    assert date in response, "Today's date not found in response"
+
+
+def test_get_question():
+    question = "What is AutoGPT and how does it compare with MetaGPT"
+
+    if cfg.SERPER_API_KEY is None:
```

Comment:
I see this test and the one above are making actual calls to 3rd party APIs. We should not do this in tests, because we may end up unknowingly hammering the 3rd party API many times.

If we're trying to test our own logic, over and above what the 3rd party API does, we should mock/stub the 3rd party API calls/responses. This will make our tests deterministic, and avoids making network calls in tests.

If we want to know whether the 3rd party API does what it claims to do, that's a different problem. We expect the 3rd party to test their own stuff.

Reply:
I'll add the `real` mark to these tests so we can skip them if necessary.

Reply:
OK. @20001LastOrder can you rename `real` to something more explicit, e.g. `external-api-caller`?